e2e-polybft/consensus_test.go

```diff
@@ -126,41 +133,97 @@ func TestE2E_Consensus_RegisterValidator(t *testing.T) {
 			ValidatorSetAddr:  contracts.ValidatorSetContract},
 		&e2eStateProvider{txRelayer: txRelayer})
 
-	// create new account
-	addrs, err := cluster.InitSecrets(newValidatorSecrets, 1)
+	// create the first account and extract the address
+	addrs, err := cluster.InitSecrets(firstValidatorDataDir, 1)
 	require.NoError(t, err)
 
-	// extract new validator address
-	newValidatorAddr := ethgo.Address(addrs[0])
+	firstValidatorAddr := ethgo.Address(addrs[0])
+
+	// create the second account and extract the address
+	addrs, err = cluster.InitSecrets(secondValidatorDataDir, 1)
+	require.NoError(t, err)
 
-	// assert that account is created
+	secondValidatorAddr := ethgo.Address(addrs[0])
+
+	// assert that accounts are created
 	validatorSecrets, err := genesis.GetValidatorKeyFiles(cluster.Config.TmpDir, cluster.Config.ValidatorPrefix)
 	require.NoError(t, err)
-	require.Equal(t, validatorSize+1, len(validatorSecrets))
+	require.Equal(t, validatorSize+2, len(validatorSecrets))
+
+	// collect owners validator secrets
+	ownerSecrets := validatorSecrets[0]
 
 	// wait for consensus to start
 	require.NoError(t, cluster.WaitForBlock(1, 10*time.Second))
 
-	// register new validator
-	require.NoError(t, srv.RegisterValidator(newValidatorSecrets, newValidatorBalanceRaw, newValidatorStakeRaw))
+	// owner enlists both new validators
+	require.NoError(t, owner.EnlistValidator(firstValidatorAddr.String(), ownerSecrets))
+	require.NoError(t, owner.EnlistValidator(secondValidatorAddr.String(), ownerSecrets))
 
-	go func() {
-		// start new validator
-		cluster.InitTestServer(t, 6, true, false)
-	}()
+	// start the first and the second validator
+	cluster.InitTestServer(t, validatorSize+1, true, false)
+	cluster.InitTestServer(t, validatorSize+2, true, false)
+
+	ownerAcc, err := sidechain.GetAccountFromDir(path.Join(cluster.Config.TmpDir, ownerSecrets))
+
+	// get the initial balance of the new validator
+	initialBalance, ok := new(big.Int).SetString(newValidatorInitBalance, 10)
+	require.True(t, ok)
+
+	// send some tokens from the owner to the first validator so that the first validator can register and stake
+	receipt1, err := txRelayer.SendTransaction(&ethgo.Transaction{
+		From:  ownerAcc.Ecdsa.Address(),
+		To:    &firstValidatorAddr,
+		Value: initialBalance,
+	}, ownerAcc.Ecdsa)
+	require.NoError(t, err)
+	require.Equal(t, uint64(types.ReceiptSuccess), receipt1.Status)
+
+	// send some tokens from the owner to the second validator so that the second validator can register and stake
+	receipt2, err := txRelayer.SendTransaction(&ethgo.Transaction{
+		From:  ownerAcc.Ecdsa.Address(),
+		To:    &secondValidatorAddr,
+		Value: initialBalance,
+	}, ownerAcc.Ecdsa)
+	require.NoError(t, err)
+	require.Equal(t, uint64(types.ReceiptSuccess), receipt2.Status)
```

Comment:
Nitpick: you can reuse the same `receipt` reference:
```suggestion
	receipt, err := txRelayer.SendTransaction(&ethgo.Transaction{
		From:  ownerAcc.Ecdsa.Address(),
		To:    &firstValidatorAddr,
		Value: initialBalance,
	}, ownerAcc.Ecdsa)
	require.NoError(t, err)
	require.Equal(t, uint64(types.ReceiptSuccess), receipt.Status)

	// send some tokens from the owner to the second validator so that the second validator can register and stake
	receipt, err = txRelayer.SendTransaction(&ethgo.Transaction{
		From:  ownerAcc.Ecdsa.Address(),
		To:    &secondValidatorAddr,
		Value: initialBalance,
	}, ownerAcc.Ecdsa)
	require.NoError(t, err)
	require.Equal(t, uint64(types.ReceiptSuccess), receipt.Status)
```

Reply:
At the beginning it was one receipt, but I would like to leave it as receipt1 and receipt2, because the code is complex and it is easier to understand it.